main/midnight/src/tme/scenario/item_character.cpp

```diff
@@ -1663,11 +1663,11 @@ namespace tme {
             defaultexport::character_t* out = (defaultexport::character_t*)data;
             VALIDATE_INFO_BLOCK(out,INFO_CHARACTER,defaultexport::character_t);
 
-        out->looking = looking;
+            out->looking = looking;
             out->time = Time();
-            out->longname = longname;
-            out->shortname = shortname;
-//            out->images =  images;
+            out->longname = std::string(longname);
+            out->shortname = std::string(shortname);
```

Comment:
Straight assignment was causing the corruption of shortname. Creating new std::string generates clean string.

Reply:
I don't think others are strings... enums, numbers, right?